ic-agent/src/agent/mod.rs

```diff
@@ -1019,6 +1021,21 @@ impl Agent {
             }
         }
     }
+
+    /// Retrieve all existing API Boundary Nodes from the State Tree.
+    pub async fn fetch_api_boundary_nodes(
+        &self,
+        effective_canister_id: Principal,
+    ) -> Result<Arc<Vec<ApiBoundaryNode>>, AgentError> {
```

Comment:
This should just be Vec<ApiBoundaryNode>. If the user wants it in an Arc they can place it there themselves.